tests/keras/preprocessing/test_image.py

```diff
@@ -118,6 +118,77 @@ def test_image_data_generator_fit(self):
         x = np.random.random((32, 3, 10, 10))
         generator.fit(x)
 
+    def test_directory_iterator(self):
+        num_classes = 2
+        tmp_folder = tempfile.mkdtemp(prefix='test_images')
+
+        # create folders and subfolders
+        paths = []
+        for cl in range(num_classes):
+            classpaths = [
+                '',
```

Comment:
Somewhat strange, maybe create the main directories first then create subdirectories?